### PR TITLE
Feature/fix initial username generation

### PR DESCRIPTION
**Pull request by Pjaijai**
That's a bad idea to trim first character from email to be username.
Update new logic to generate initial username with email and uuid.

### CODE REVIEW

File: supabase/migrations/20231005012958_fix_hanlde_new_user_function.sql [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  -- Extract the username from the email (word before @), limited to 4 characters
+  username_text := SUBSTRING(NEW.email FROM 1 FOR POSITION('@' IN NEW.email) - 1);
+  IF LENGTH(username_text) > 4 THEN
+    username_text := LEFT(username_text, 4);
+  END IF;
+
+  -- Ensure the username doesn't exceed 10 characters in total
+  IF LENGTH(username_text) + 4 > 10 THEN
+    username_text := LEFT(username_text, 10 - 4);
+  END IF;
```
Comment (yaucp) [inline supabase/migrations/20231005012958_fix_hanlde_new_user_function.sql:21]: not sure why is this here? why not just combine this with the last if?

Comment (Pjaijai) [reply supabase/migrations/20231005012958_fix_hanlde_new_user_function.sql]: Updated

**Review by yaucp — APPROVED** [review]
